streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -246,9 +246,7 @@ class StreamingContext private[streaming] (
     checkpointDir != null
   }
 
-  private[streaming] def initialCheckpoint: Checkpoint = {
-    if (isCheckpointPresent) _cp else null
-  }
+  private[streaming] def initialCheckpoint: Checkpoint = _cp
```

Comment:
Yes it's functionally equivalent. The method isn't overridden. I suppose that it's conceivable the logic of `isCheckpointPresent` could change later though. Other methods have a similar check too. Although it's a no-op change and small simplification, it doesn't matter in practice to check "if (a != null) a else null" Since the purpose here is just typo fixes, I'd stick to that and not bother with this change.